src/exchanges/adapters/ZeroExV2Adapter.sol

```diff
@@ -131,47 +146,32 @@ contract ZeroExV2Adapter is ExchangeAdapter, OrderTaker {
 
     /// @notice Validate the parameters of a takeOrder call
     /// @param _targetExchange Address of 0x v2 exchange
-    /// @param _orderAddresses [0] 0x Order param: makerAddress
-    /// @param _orderAddresses [1] 0x Order param: takerAddress
-    /// @param _orderAddresses [2] Maker asset
-    /// @param _orderAddresses [3] Taker asset
-    /// @param _orderAddresses [4] 0x Order param: feeRecipientAddress
-    /// @param _orderAddresses [5] 0x Order param: senderAddress
-    /// @param _orderData [0] 0x Order param: makerAssetData
-    /// @param _orderData [1] 0x Order param: takerAssetData
-    /// @param _orderData [2] 0x Order param: makerFeeAssetData
-    /// @param _orderData [3] 0x Order param: takerFeeAssetData
-    /// @param _orderValues [0] 0x Order param: makerAssetAmount
-    /// @param _orderValues [1] 0x Order param: takerAssetAmount
-    /// @param _orderValues [2] 0x Order param: makerFee
-    /// @param _orderValues [3] 0x Order param: takerFee
-    /// @param _orderValues [4] 0x Order param: expirationTimeSeconds
-    /// @param _orderValues [5] 0x Order param: salt
-    /// @param _orderValues [6] Taker asset fill quantity
-    /// @param _identifier Order identifier
-    /// @param _signature Signature of the order
+    /// @param _encodedArgs Encoded parameters passed from client side
     function __validateTakeOrderParams(
         address _targetExchange,
-        address[8] memory _orderAddresses,
-        uint256[8] memory _orderValues,
-        bytes[4] memory _orderData,
-        bytes32 _identifier,
-        bytes memory _signature
+        bytes memory _encodedArgs
     )
         internal
         view
         override
     {
-        require(
-            __getAssetAddress(_orderData[0]) == _orderAddresses[2],
-            "__validateTakeOrderParams: makerAssetData does not match address"
+        (
+            ,
+            uint256[7] memory orderValues,
+            bytes [2] memory orderData
+            ,
+        ) = __decodeTakeOrderArgs(_encodedArgs);
+
+        IRegistry registry = __getRegistry();
+        require(registry.assetIsRegistered(
+            __getAssetAddress(orderData[0])), 'Maker asset not registered'
         );
-        require(
-            __getAssetAddress(_orderData[1]) == _orderAddresses[3],
-            "__validateTakeOrderParams: takerAssetData does not match address"
+        require(registry.assetIsRegistered(
+            __getAssetAddress(orderData[1])), 'Taker asset not registered'
```

Comment:
It doesn't matter too much, but @travs and I started to move to this pattern of "functionName: reason" for revert strings, e.g., "__validateTakeOrderParams: takerAssetData does not match address". It's not a huge deal, but you can apply this to this PR as well if you'd like.